### PR TITLE
Patched Dolma, added twemojis to chat and implemented Dolma in Kibbeh

### LINKED ISSUE #2660: I want the message rendering done differently
## So what's happening right now
Currently, on message send, we're sending "blocks" of data (link,text,emote) and rendering these blocks once a message is received by the end user.

*This will make `dolma` obsolete*

## What I want to happen
I want to get rid of the blocks stage and do the entire message parsing on the end user side instead of the sender. It will allow us for more performant rendering (for example, instead of adding a ParseTwemoji element to each block of text, we could parse it all at once)

@benawad

### PR DESCRIPTION
**Pull request by goldyydev**
@benawad **for it to work:**
- please push #2762 to prod
- fix this vercel deploy
image

### CODE REVIEW

**Comment by goldyydev** [comment]
@benawad you need to modify the deploy/test tasks

**Comment by ghost** [comment]
Amitoj fixed it i think

**Comment by amitojsingh366** [comment]
@goldyydev i just fixed the tests, pull latest and push

File: dolma/src/lib/decode.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-	const tokens = encodeTokens(all);
+export function decodeTokens(this: {emotes: {name: string}[]}, all: Array<Unitoken | MessageToken | string> | string): string {
+	const tokens = new dolma(this.emotes).encode(all);
 	let vals: string[] = [];
 
-	tokens.map(tkn => {
+	tokens.tokens.map(tkn => {
 		if (tkn.t == 'text') return vals.push(tkn.v);
 		if (tkn.t == 'block') return vals.push(`\`${tkn.v}\``);
 		if (tkn.t == 'emote') return vals.push(`:${tkn.v}:`);
```
Comment (amitojsingh366) [inline dolma/src/lib/decode.ts:13]: please add `emoji` token to decoding aswell

File: dolma/src/tokens/types/emoji.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Token } from "../../util/types/tokenTypes";
+import emojiRegex from "emoji-regex"
+
+export default {
+	name: "emote",
```
Comment (amitojsingh366) [inline dolma/src/tokens/types/emoji.ts]: the name for this token is `emoji` not `emote`

**Review by amitojsingh366 — CHANGES_REQUESTED** [review]

File: dolma/src/lib/decode.ts (hunk 1 above)
Comment (goldyydev) [reply dolma/src/lib/decode.ts:13]: the decode function is broken, it will be fixed in a separate pr

File: dolma/src/tokens/types/emoji.ts (hunk 2 above)
Comment (goldyydev) [reply dolma/src/tokens/types/emoji.ts]:
```suggestion
	name: "emoji",
```

File: CODEOWNERS [hunk 3]
```diff
@@ -1,4 +1,4 @@
 /kebab/ @overlisted
 /baklava/ @amitojsingh366
-/dolma/ @HoloPanio
+/dolma/ @HoloPanio @goldyydev
```
Comment (amitojsingh366) [inline CODEOWNERS]: i don't think u should add yourself to code owners, leave the final dolma authority to @HoloPanio as he made it

**Review by amitojsingh366 — CHANGES_REQUESTED** [review]

**Comment by amitojsingh366** [comment]
please follow the proper pattern for commit messages *(💡TIP: you can use `yarn commit` or `yarn commit:signed` and answer the questions asked to generate a properly formatted one for you!)*

to fix your most recent commit you can run the command `git commit --amend -m "properly formatted commit message here"`

to fix your older/multiple commits please take a look at [this article](https://docs.github.com/en/github/committing-changes-to-your-project/changing-a-commit-message#amending-older-or-multiple-commit-messages)

**Review by HoloPanio — APPROVED** [review]

**Pull request merged**